minimum viable product

ben-spector-cc opened the pull request:
Some design patterns that I'm using that I'd love to hear your thoughts on:

* I'm using a constant naming pattern for the message types. The pattern is `SOURCE_MODE.TYPE`.
  * `SOURCE` can be either `CLIENT` or `SERVER`
  * `MODE` can be either `MESSAGE` (1:1) or `BROADCAST`(1:many) 
  * `TYPE` will be unique to the message type 
  * For example: `SERVER_BROADCAST.COUNTDOWN` or `CLIENT_MESSAGE.NEW_USER`

* To implement custom messages, I send all data between the server and client as a `JSON.stringify`-ed object containing a required `.type` property. If the message needs to include a payload, there's an optional `.payload` property that I have consistently made an object containing the data, even if the data is a single primitive value. 
  * For example: `JSON.stringify({ type: CLIENT_MESSAGE.NEW_USER, payload: { name: 'Ben' } })`

Aside from these patterns, any feedback you have on the WebSockets implementation, the application structure, and the application's organization would be great. 

And for now, please focus your feedback on the parts of the application related to what the user will have to implement. Please ignore the DOM logic and we can circle back to it once we get the final designs in from our designer.

Thanks!

ben-spector-cc committed:
minimum viable product

ben-spector-cc committed:
fixed new potato holder bug

Ethan-Arrowood commented on public/index.html:
```diff
@@ -0,0 +1,182 @@
+  </div>
+    
+  <script>
+    (function() {
```
iife is not necessary
```suggestion

```

Ethan-Arrowood commented on public/index.html:
```diff
@@ -0,0 +1,182 @@
+      }
+  
+      init();
+    })();
```
```suggestion

```

Ethan-Arrowood commented on server.js:
```diff
@@ -0,0 +1,136 @@
+  GAME_OVER: 'GAME_OVER',
+  GAME_START: 'GAME_START',
+}
+
```
```suggestion
const CONSTANTS = {
  CLIENT: {
    MESSAGE: {
      NEW_USER: 'new user',
      PASS_POTATO: 'pass potato'
    }
  },
  SERVER: {
    MESSAGE: {
      PLAYER_ASSIGNMENT: 'player assignment',
      GAME_FULL: 'game full',
    },
    BROADCAST: {
      COUNTDOWN: 'COUNTDOWN',
      NEW_POTATO_HOLDER: 'NEW_POTATO_HOLDER',
      GAME_OVER: 'GAME_OVER',
      GAME_START: 'GAME_START',
    }
  }
}
```

Ethan-Arrowood commented on server.js:
```diff
@@ -0,0 +1,136 @@
+}
+
+// Application Variables;
+let players = [];
```
```suggestion
const players = [];
```

ben-spector-cc committed:
refactored constants

ben-spector-cc commented:
@Ethan-Arrowood Just moved the constants over into their own file and removed the IIFE. I also refactored the HTTP server request/response handler to be a little bit cleaner.

Ethan-Arrowood reviewed (approved):
lgtm
